src/main/java/com/klodnicki/Bike/v2/service/UserService.java

```diff
@@ -1,7 +1,71 @@
 package com.klodnicki.Bike.v2.service;
 
+import com.klodnicki.Bike.v2.DTO.user.UserForAdminResponseDTO;
+import com.klodnicki.Bike.v2.model.entity.User;
+import com.klodnicki.Bike.v2.repository.UserRepository;
+import com.klodnicki.Bike.v2.service.interfacee.GenericUserService;
+import lombok.AllArgsConstructor;
+import org.modelmapper.ModelMapper;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Service;
 
+import java.util.ArrayList;
+import java.util.List;
+
 @Service
-public class UserService {
+@AllArgsConstructor
+public class UserService implements GenericUserService {
+
+    private final UserRepository userRepository;
+    private final ModelMapper modelMapper;
+
+    @Override
+    public UserForAdminResponseDTO add(User user) {
+        User user1 = userRepository.save(user);
+
+        return modelMapper.map(user1, UserForAdminResponseDTO.class);
+    }
+    @Override
+    public List<UserForAdminResponseDTO> findAll() {
+        Iterable<User> users = userRepository.findAll();
+        List<UserForAdminResponseDTO> listUsersDTO = new ArrayList<>();
+
+        for (User user: users) {
+            UserForAdminResponseDTO userDto = modelMapper.map(user, UserForAdminResponseDTO.class);
+            listUsersDTO.add(userDto);
+        }
+        return listUsersDTO;
+    }
+    @Override
+    public UserForAdminResponseDTO findById(Long id) {
+        User user = getUser(id);
+
+        return modelMapper.map(user, UserForAdminResponseDTO.class);
+    }
+    @Override
+    public void deleteById(Long id) {
+        userRepository.deleteById(id);
+    }
+    @Override
+    public User findUserById(Long id) {
+        return userRepository.findById(id).orElseThrow(IllegalArgumentException::new);
+    }
+
+    private User getUser(Long id) {
+        return userRepository.findById(id).orElseThrow(IllegalArgumentException::new);
+    }
```

Comment:
What is the difference between both those methods? You can remove the bottom one.
```suggestion
    @Override
    public User findUserById(Long id) {
        return userRepository.findById(id).orElseThrow(IllegalArgumentException::new);
    }

```

Reply:
you are right! thanks!